application/src/main/java/org/opentripplanner/updater/trip/gtfs/GtfsRealTimeTripUpdateAdapter.java

```diff
@@ -679,25 +707,59 @@ private Result<UpdateSuccess, UpdateError> handleAddedTrip(
       debug(
         tripId,
         serviceDate,
-        "ADDED trip has service date {} for which no service id is available, skipping.",
+        "NEW trip has service date {} for which no service id is available, skipping.",
         serviceDate.toString()
       );
       return UpdateError.result(tripId, NO_SERVICE_ON_DATE);
     } else {
       // Just use first service id of set
       tripBuilder.withServiceId(serviceIds.iterator().next());
     }
-    return addTripToGraphAndBuffer(
+
+    var tripHeadsign = getTripHeadsign(tripUpdate);
```

Comment:
Because this class is already to long, this should go into a new wrapper object around the raw TripUpdate so we can separate the getters from the business logic. Return optionals if possible.